zipline/finance/trading.py

```diff
@@ -69,7 +70,7 @@ def __init__(
         bm_symbol='^GSPC',
         exchange_tz="US/Eastern",
         max_date=None,
-        env_trading_calendar=tradingcalendar,
+        env_trading_calendar=NYSEExchangeCalendar,
```

Comment:
Good point about the calendar arguments. Were you thinking instead of instantiating the `NYSEExchangeCalendar` outside of the `TradingEnvironment`, and then passing it in? That doesn't seem ideal to me, since it implies that the `NYSEExchangeCalendar` would be instantiated every time `trading.py` was imported. How about we add `min_date` and `max_date` kwargs to the `TradingEnvironment`, as Scotty did here: https://github.com/quantopian/zipline/pull/556/files#diff-84720364fb47256a081315e8449ee0d2R101